### PR TITLE
PHPCS: admin pages

### LINKED ISSUE #21948: PHPCS: jetpack/_inc/lib/admin-pages/
### Impacted plugin

Jetpack

### Steps to Reproduce

```
	"projects/plugins/jetpack/_inc/lib/admin-pages/class.jetpack-admin-page.php",
	"projects/plugins/jetpack/_inc/lib/admin-pages/class.jetpack-landing-page.php",
	"projects/plugins/jetpack/_inc/lib/admin-pages/class.jetpack-react-page.php",
	"projects/plugins/jetpack/_inc/lib/admin-pages/class.jetpack-settings-page.php",
```

### A clear and concise description of what you expected to happen.

_No response_

### What actually happened

_No response_

### Other information

_No response_

### Operating System

_No response_

### OS Version

_No response_

### Browser

Chrome/Chromium

### Browser Version(s)

_No response_

### PR DESCRIPTION
**Pull request by bindlegirl**
Fixes #21948

#### Changes proposed in this Pull Request:

* Mostly cleared simple errors
* Added wp_kses() in class.jetpack-settings-page
* Replaced the file_get_contents with WP_Filesystem 

#### Jetpack product discussion

n/a

#### Does this pull request change what data or activity we track or use?

no

#### Testing instructions:

* Test all admin pages and see if everything shows up.

### CODE REVIEW

File: projects/plugins/jetpack/_inc/lib/admin-pages/class.jetpack-admin-page.php [hunk 1]
```diff
@@ -1,36 +1,52 @@
...
 use Automattic\Jetpack\Status;
 
-// Shared logic between Jetpack admin pages
+/**
+ * Shared logic between Jetpack admin pages.
+ */
 abstract class Jetpack_Admin_Page {
-	// Add page specific actions given the page hook
-	abstract function add_page_actions( $hook );
+	/** Add page specific actions given the page hook.
```
Comment (anomiex) [inline projects/plugins/jetpack/_inc/lib/admin-pages/class.jetpack-admin-page.php:16]: We normally don't start the summary on the same line as the `/**`.
```suggestion
	/**
	 * Add page specific actions given the page hook.
```

File: projects/plugins/jetpack/_inc/lib/admin-pages/class.jetpack-admin-page.php [hunk 2]
```diff
@@ -1,36 +1,52 @@
...
+	 *
+	 * @param string $hook Hook of current page.
+	 */
+	abstract public function add_page_actions( $hook );
 
-	// Create a menu item for the page and returns the hook
-	abstract function get_page_hook();
+	/**
+	 * Create a menu item for the page and returns the hook.
```
Comment (anomiex) [inline projects/plugins/jetpack/_inc/lib/admin-pages/class.jetpack-admin-page.php:23]: Should document that return value.
```suggestion
	 * Create a menu item for the page and returns the hook.
	 *
	 * @return string|false Return value from WordPress's `add_menu_page()` or `add_submenu_page()`.
```

File: projects/plugins/jetpack/_inc/lib/admin-pages/class.jetpack-admin-page.php [hunk 3]
```diff
@@ -51,8 +67,10 @@ public function on_jetpack_loaded( $jetpack ) {
 			Identity_Crisis::validate_sync_error_idc_option() && ! Jetpack_Options::get_option( 'safe_mode_confirmed' )
 		);
 	}
-
-	function add_actions() {
+	/**
```
Comment (anomiex) [inline projects/plugins/jetpack/_inc/lib/admin-pages/class.jetpack-admin-page.php:70]: It seems clearer when reading the file if there's a blank line between the end of the previous method and the next method's doc block.
```suggestion

	/**
```

File: projects/plugins/jetpack/_inc/lib/admin-pages/class.jetpack-admin-page.php [hunk 4]
```diff
@@ -116,38 +134,49 @@ private function add_connection_banner_actions() {
...
+	public function render() {
 		// We're in an IDC: we need a decision made before we show the UI again.
 		if ( self::$block_page_rendering_for_idc ) {
 			return;
 		}
 
-		// Check if we are looking at the main dashboard
-		if ( isset( $_GET['page'] ) && 'jetpack' === $_GET['page'] ) {
+		// Check if we are looking at the main dashboard.
+		if ( isset( $_GET['page'] ) && 'jetpack' === $_GET['page'] ) {// phpcs:ignore WordPress.Security.NonceVerification.Recommended
```
Comment (anomiex) [inline projects/plugins/jetpack/_inc/lib/admin-pages/class.jetpack-admin-page.php:147]: We normally have a space before the `//`, and it's good practice to comment why we're ignoring the nonce sniff.
```suggestion
		if ( isset( $_GET['page'] ) && 'jetpack' === $_GET['page'] ) { // phpcs:ignore WordPress.Security.NonceVerification.Recommended -- View logic.
```

File: projects/plugins/jetpack/_inc/lib/admin-pages/class.jetpack-admin-page.php [hunk 5]
```diff
@@ -116,38 +134,49 @@ private function add_connection_banner_actions() {
...
 			$this->page_render();
 			return;
 		}
 		self::wrap_ui( array( $this, 'page_render' ) );
 	}
-
-	function admin_help() {
+	/**
```
Comment (anomiex) [inline projects/plugins/jetpack/_inc/lib/admin-pages/class.jetpack-admin-page.php:153]:
```suggestion

	/**
```

File: projects/plugins/jetpack/_inc/lib/admin-pages/class.jetpack-admin-page.php [hunk 6]
```diff
@@ -116,38 +134,49 @@ private function add_connection_banner_actions() {
...
+	 *
+	 * @todo This may no longer be used.
+	 */
+	public function admin_help() {
 		$this->jetpack->admin_help();
 	}
-
-	function admin_page_load() {
-		// This is big.  For the moment, just call the existing one.
+	/**
```
Comment (anomiex) [inline projects/plugins/jetpack/_inc/lib/admin-pages/class.jetpack-admin-page.php:161]:
```suggestion

	/**
```

File: projects/plugins/jetpack/_inc/lib/admin-pages/class.jetpack-admin-page.php [hunk 7]
```diff
@@ -174,11 +203,11 @@ function is_rest_api_enabled() {
 	 *
 	 * @since 4.4.0
 	 *
-	 * @param $page
+	 * @param Page $page string Pages related to paid modules.
```
Comment (anomiex) [inline projects/plugins/jetpack/_inc/lib/admin-pages/class.jetpack-admin-page.php:206]: Looks like this function is used with the `current_screen` action, which describes the parameter as
```suggestion
	 * @param WP_Screen $page Current WP_Screen object.
```

File: projects/plugins/jetpack/_inc/lib/admin-pages/class.jetpack-admin-page.php [hunk 8]
```diff
@@ -230,8 +260,10 @@ function check_plan_deactivate_modules( $page ) {
 			'deactivate' => $to_deactivate,
 		);
 	}
-
-	static function load_wrapper_styles() {
+	/**
```
Comment (anomiex) [inline projects/plugins/jetpack/_inc/lib/admin-pages/class.jetpack-admin-page.php:263]:
```suggestion

	/**
```

File: projects/plugins/jetpack/_inc/lib/admin-pages/class.jetpack-admin-page.php [hunk 9]
```diff
@@ -277,7 +309,12 @@ static function load_wrapper_styles() {
 		';
 		wp_add_inline_style( 'dops-css', $custom_css );
 	}
-
+	/**
+	 * Build footer, content, and footer for admin page.
+	 *
+	 * @param string $callback The callback sent to the Jetpack_Admin_Page::wrap_ui method. Callback is responsible for any needed escaping.
+	 * @param array  $args The arguments sent to the Jetpack_Admin_Page::wrap_ui method.
```
Comment (anomiex) [inline projects/plugins/jetpack/_inc/lib/admin-pages/class.jetpack-admin-page.php:316]: These descriptions aren't very descriptive. Maybe something like this instead?
```suggestion
	 * @param string $callback Callback to produce the content of the page. The callback is responsible for any needed escaping.
	 * @param array  $args Options for the wrapping. Also passed to the `jetpack_admin_pages_wrap_ui_after_callback` action.
	 *   - is-wide: (bool) Set the "is-wide" class on the wrapper div, which increases the max width. Default false.
	 *   - show-nav: (bool) Whether to show the navigation bar at the top of the page. Default true.
```

File: projects/plugins/jetpack/_inc/lib/admin-pages/class.jetpack-react-page.php [hunk 10]
```diff
@@ -1,43 +1,62 @@
...
-// Builds the landing page and its menu
+/**
+ * Builds the landing page and its menu.
+ */
 class Jetpack_React_Page extends Jetpack_Admin_Page {
-
+	/**
+	 * Show the landing page only when Jetpack is connected.
+	 *
+	 * @var boolean
```
Comment (anomiex) [inline projects/plugins/jetpack/_inc/lib/admin-pages/class.jetpack-react-page.php:16]: It seems we normally use "bool" rather than "boolean" in PHP.
```suggestion
	 * @var bool
```

File: projects/plugins/jetpack/_inc/lib/admin-pages/class.jetpack-react-page.php [hunk 11]
```diff
@@ -1,43 +1,62 @@
...
+	 */
 	protected $dont_show_if_not_active = false;
 
+	/**
+	 * Used for fallback when REST API is disabled.
+	 *
+	 * @var boolean
```
Comment (anomiex) [inline projects/plugins/jetpack/_inc/lib/admin-pages/class.jetpack-react-page.php:23]:
```suggestion
	 * @var bool
```

File: projects/plugins/jetpack/_inc/lib/admin-pages/class.jetpack-react-page.php [hunk 12]
```diff
@@ -1,43 +1,62 @@
...
+	 */
 	protected $is_redirecting = false;
 
-	function get_page_hook() {
-		// Add the main admin Jetpack menu
+	/**
+	 * Add the main admin Jetpack menu.
```
Comment (anomiex) [inline projects/plugins/jetpack/_inc/lib/admin-pages/class.jetpack-react-page.php:28]:
```suggestion
	 * Add the main admin Jetpack menu.
	 *
	 * @return string|false Return value from WordPress's `add_menu_page()`.
```

File: projects/plugins/jetpack/_inc/lib/admin-pages/class.jetpack-react-page.php [hunk 13]
```diff
@@ -1,43 +1,62 @@
...
+	 * @param string $hook Hook of current page.
+	 * @return void
+	 */
+	public function add_page_actions( $hook ) {
 		/** This action is documented in class.jetpack-admin.php */
 		do_action( 'jetpack_admin_menu', $hook );
 
-		if ( ! isset( $_GET['page'] ) || 'jetpack' !== $_GET['page'] ) {
-			return; // No need to handle the fallback redirection if we are not on the Jetpack page
+		if ( ! isset( $_GET['page'] ) || 'jetpack' !== $_GET['page'] ) {// phpcs:ignore WordPress.Security.NonceVerification.Recommended
```
Comment (anomiex) [inline projects/plugins/jetpack/_inc/lib/admin-pages/class.jetpack-react-page.php:44]:
```suggestion
		if ( ! isset( $_GET['page'] ) || 'jetpack' !== $_GET['page'] ) { // phpcs:ignore WordPress.Security.NonceVerification.Recommended -- This is view logic.
```

File: projects/plugins/jetpack/_inc/lib/admin-pages/class.jetpack-settings-page.php [hunk 14]
```diff
@@ -42,25 +61,25 @@ function page_render() {
 		);
 		$noscript_notice = str_replace(
 			'#TEXT#',
-			esc_html__( "Turn on JavaScript to unlock Jetpack's full potential!", 'jetpack' ),
+			esc_html( __( "Turn on JavaScript to unlock Jetpack's full potential!", 'jetpack' ) ),
```
Comment (anomiex) [inline projects/plugins/jetpack/_inc/lib/admin-pages/class.jetpack-settings-page.php:64]: Why did you change this one from `esc_html__()` to `esc_html( __() )`, and the two below in the opposite direction?

File: projects/plugins/jetpack/_inc/lib/admin-pages/class.jetpack-settings-page.php [hunk 15]
```diff
@@ -109,7 +165,7 @@ function page_render() {
 					</div>
 					<div class="manage-left" style="width: 100%;">
 						<form class="jetpack-modules-list-table-form" onsubmit="return false;">
-						<table class="<?php echo implode( ' ', $list_table->get_table_classes() ); ?>">
+						<table class="<?php echo implode( ' ', $list_table->get_table_classes() ); // phpcs:ignore WordPress.Security.EscapeOutput.OutputNotEscaped -- Escaped in class.jetpack-modules-list-table.php ?>">
```
Comment (anomiex) [inline projects/plugins/jetpack/_inc/lib/admin-pages/class.jetpack-settings-page.php:168]: There's no escaping there. There isn't really any need for it either, but I'd go ahead and do `esc_attr()` here anyway.
```suggestion
						<table class="<?php echo esc_attr( implode( ' ', $list_table->get_table_classes() ) ); ?>">
```

**Review by anomiex — COMMENTED** [review]
Sorry for all the whitespace nitpicks.

File: projects/plugins/jetpack/_inc/lib/admin-pages/class.jetpack-admin-page.php (hunk 9 above)
Comment (bindlegirl) [reply projects/plugins/jetpack/_inc/lib/admin-pages/class.jetpack-admin-page.php:317]: Sounds much better :-)

File: projects/plugins/jetpack/_inc/lib/admin-pages/class.jetpack-settings-page.php (hunk 14 above)
Comment (bindlegirl) [reply projects/plugins/jetpack/_inc/lib/admin-pages/class.jetpack-settings-page.php:64]: Uh, that was a mistake that happened while I was experimenting with some alternative ways of escaping this. Thanks for catching this.

Comment (bindlegirl) [reply projects/plugins/jetpack/_inc/lib/admin-pages/class.jetpack-settings-page.php]: Fixed this in fbe1c74fe3df5767e14c464f13769a53efac862b
